src/raster/rasterband.rs

```diff
@@ -863,6 +954,50 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    n_buckets: i32,
```

Comment:
Why isn't this `usize`? Sometimes GDAL uses signed values when it doesn't make semantic sense, and there's [precedence](https://github.com/georust/gdal/blob/f180427c099f0093f7b8b10111b556517ee7cba9/src/raster/rasterband.rs#L83-L85) for rectifying this in Rust.

Reply:
I haven't contributed for a while and from memory we were using `isize`. I do agree that we could rectify that.

How far do we want to take it? I could also use a `NonZeroUsize`, which could be also useful when selecting a band from a dataset... (@lnicola)

Reply:
Lol...  yeah, I've argued for `NonZeroUsize`, but creating them is really annoying/verbose.

When I see `isize` or `i32` I immediately ask myself "What does a negative value mean?". If it has meaning, we should document it. If it doesn't, then I'd at least try in the bindings to coerce it into an unsigned type.

Reply:
Yeah, I wish constructing a `NonZeroUsize` from a constant/literal was possible without `unsafe` or bloating the binary. :pensive: